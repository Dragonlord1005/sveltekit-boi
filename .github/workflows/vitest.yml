name: Vitest Workflow

on:
  push:
    branches: [ main ]

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install Node.js
      uses: actions/setup-node@v3
      with:
        node-version: 16

    - name: Install pnpm
      uses: pnpm/action-setup@v2
      with:
        version: latest

    - name: Install dependencies
      run: pnpm install

    - name: Run tests
      run: pnpm exec vitest run

    # - name: Check exit code
    #   if: ${{ failure() }}
    #   run: exit 1