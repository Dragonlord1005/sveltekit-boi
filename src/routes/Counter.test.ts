import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/svelte';
import Counter from './Counter.svelte';
import userEvent from '@testing-library/user-event';

describe('Counter', () => {
    it('should render', () => {
        render(Counter);
        expect(Counter).toMatchSnapshot();
    });

    it('should be able to increment', async () => {
        const user = userEvent.setup()

        const { component } = render(Counter);

        // Mock the increment function
        let count = 0;
        const mock = vi.fn((event) => {
            count = event.detail.count;
            console.log('count', count);
        });
        component.$on('increment', mock);

        const button = screen.getByRole('button', { name: /increase the counter by one/i });
        await user.click(button);

        // expect(mock).toHaveBeenCalled();
        expect(count).toBe(1);

    });
}
);