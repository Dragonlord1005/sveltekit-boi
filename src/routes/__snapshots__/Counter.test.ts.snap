// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter > should render 1`] = `[Function]`;
